### PR TITLE
build: remove workaround for python fallback

### PR DESCRIPTION
**Pull request by compnerd**
This forces the python3 interpreter to be found and no longer permits
the use of Python2 as a fallback if python3 is not found.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @JDevlieghere @shahmishal @drexin

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by shahmishal** [comment]
@swift-ci test

**Comment by shahmishal** [comment]
@swift-ci build toolchain

**Comment by tbkka** [comment]
You should make the same change to `cmake/modules/StandaloneOverlay.cmake`

**Review by tbkka — APPROVED** [review]
I believe Mishal wants us to wait until next week to merge these changes (so he has time to double-check some of the CI configurations).

I think we should re-run tests on this PR every day or so to make sure it remains stable until we're ready to merge.

**Comment by tbkka** [comment]
@swift-ci Please test

**Review by shahmishal — APPROVED** [review]
LGTM! Feel free to merge this when you'r ready!

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**
